### PR TITLE
Deleting inexistent user or expression: response and tests

### LINKED ISSUE #17: Response and tests for deleting inexistent user or expression

### PR DESCRIPTION
**Pull request by RazvanKokovics**
Closes #17. Should be merged after #16.

### CODE REVIEW

**Comment by gasher** [comment]
Please use a better name for this PR

File: service/expression.js [hunk 1]
```diff
@@ -25,8 +26,13 @@ class ExpressionService {
...
   async removeExpression(expressionId) {
-    await expressionRepository.deleteExpression(expressionId);
-    await expressionRepository.deleteExpressionFromUsers(expressionId);
+    const deleted = await expressionRepository.deleteExpression(expressionId);
+
+    if (!deleted) {
+      throw new InexistentItem('ExpressionId does not exists.');
+    }
+
+    expressionRepository.deleteExpressionFromUsers(expressionId);
```
Comment (gasher) [inline service/expression.js]: I might be wrong but shouldn't you await this or return it to be awaited in the controller?

File: service/users.js [hunk 2]
```diff
@@ -19,7 +19,13 @@ class UserService {
...
 
-  removeUser(userName) {
+  async removeUser(userName) {
+    const user = await userRepository.getUserByUserName(userName);
+
+    if (!user) {
+      throw new InexistentItem('Username does not exists.');
+    }
+
     userRepository.deleteUser(userName);
```
Comment (gasher) [inline service/users.js]: Here as well. I dont think you are await-ing it in the repository so the promise wont be await-ed at all

**Review by gasher — CHANGES_REQUESTED** [review]
I saw some issues that might be related to this PR

File: service/expression.js [hunk 3]
```diff
@@ -26,8 +27,13 @@ class ExpressionService {
   }
 
   async removeExpression(expressionId) {
-    await expressionRepository.deleteExpression(expressionId);
-    await expressionRepository.deleteExpressionFromUsers(expressionId);
+    const deleted = await expressionRepository.deleteExpression(expressionId);
+
+    if (!deleted) {
+      throw new InexistentItem('ExpressionId does not exists.');
```
Comment (gasher) [inline service/expression.js]: This message is quite confusing. I think you are talking about an expression with that id that doesn't exists.

File: service/users.js [hunk 4]
```diff
@@ -19,8 +19,14 @@ class UserService {
     return userRepository.addUser(userWithHashedPassword);
   }
 
-  removeUser(userName) {
-    userRepository.deleteUser(userName);
+  async removeUser(userName) {
+    const user = await userRepository.getUserByUserName(userName);
+
+    if (!user) {
+      throw new InexistentItem('Username does not exists.');
```
Comment (gasher) [inline service/users.js]: Same here, the user with that username doesn't exists

**Comment by RazvanKokovics** [comment]
This PR was re-made in #25 .

**Pull request closed without merging**